### DISCUSSION TITLE
Reinstalled VisualStudioCode, can not compile Tasmota versions

### DISCUSSION THREAD

**Discussion by wchulze** [arendst/Tasmota#22043] [Q&A]
I had  reinstalled VisualStudioCode on Windows 10. To test it, I compiled all my Tasmota versions. Unfortunately, it caused problems with version 8.5.1 

I got the error message:

**Tool Manager: Installing jason2866/framework-arduinoespressif8266 @ *
UnknownPackageError: Could not find the package with 'jason2866/framework-arduinoespressif8266 @ *' requirements for your system 'windows_amd64'**

As I am still using the version 8.5.1 I would like to continue to maintain it. Where can I get the missing package and how can I install it?

**Comment by Jason2866** [comment]
Platformio removed old outdated frameworks from there registry. I may can help when you know the used Arduino core.

**Comment by wchulze** [comment]
Thank*s for quick reply.

Here is the entry from platform.ini

```
[core]
; *** Esp8266 Tasmota modified Arduino core based on core 2.7.4
platform                  = espressif8266@2.6.2
platform_packages         = jason2866/framework-arduinoespressif8266
build_unflags             = ${esp_defaults.build_unflags}
build_flags               = ${esp82xx_defaults.build_flags}
```

I hope you can do something with it. I have no idea how to get the package and then install it.

**Comment by Jason2866** [comment]
Changing to
```
[core]
; *** Esp8266 Tasmota modified Arduino core based on core 2.7.4. Added Backport for PWM selection
platform                    = https://github.com/tasmota/platform-espressif8266/releases/download/2024.06.00/platform-espressif8266.zip
platform_packages           =
build_unflags               = ${esp_defaults.build_unflags}
build_flags                 = ${esp82xx_defaults.build_flags}
; *** Use ONE of the two PWM variants. Tasmota default is Locked PWM
                              ;-DWAVEFORM_LOCKED_PHASE
                              -DWAVEFORM_LOCKED_PWM
```
should work. There are no relevant changes in esp8266 since that time.

**Reply by wchulze** [reply]
Thks, it compiles with framework 2.7.7 but chrash with

```
Compiling .pio\build\tasmota-DE\lib90c\ESP8266Audio\AudioOutputSTDIO.cpp.o
lib\ESP8266Audio\src\AudioOutputSPIFFSWAV.cpp: In member function 'virtual bool AudioOutputSPIFFSWAV::begin()':
lib\ESP8266Audio\src\AudioOutputSPIFFSWAV.cpp:46:3: error: 'SPIFFS' was not declared in this scope
   SPIFFS.remove(filename);
   ^
*** [.pio\build\tasmota-DE\lib90c\ESP8266Audio\AudioOutputSPIFFSWAV.cpp.o] Error 1
```

**Reply by Jason2866** [reply]
Yeah, that's one of the few changes. You have to remove the SPIFFS code parts.
Use the lib from actual Tasmota. Honestly I don't understand why you are want to compile old Tasmota.

**Reply by wchulze** [reply]
This is easy to explain: I am using version 8.5.1 and below, by adding added some of my own extensions in the past. Sometimes I want to make a change. Besides, the version runs efficiently and there are no worries about Sonoff products.
The disposal of the SPIFFS code is unclear to me, as it is not part of the 8.5.1 version.

**Reply by Jason2866** [reply]
SPIFFS is removed from Tasmota Arduino Core, since SPIFFS is not used in Tasmota.
Used libs have all adopted to compile without SPIFFS. This was not the case with v8.5.1.
So when you want to compile the v8.5.1 now you have to make some changes yourself, since the old esp8266 framework is not available anymore.

**Reply by wchulze** [reply]
I would like to make the changes to the software. But the compiler error message does not give any information about how the error actually occurs. What should I do to fix it? Where does the old software reference this part of the SPIFFS so that it can be cleaned up?

**Reply by Jason2866** [reply]
There is NO SPIFFS code in the Arduino esp8266 core anymore. You have to remove the reference to SPIFFS   in any code or library in Tasmota.

**Reply by wchulze** [reply]
Thank you for all your efforts and tips.

I have removed all references to the SPIFFS code. These were only found in the AudioOutputSPIFFSWAV.cpp module. I have corrected the code so that the lib is no longer called and the corresponding code is not included, bingo compiled without error. I flashed the image into an old Sonoff-Basic R2 from 2017. No problems!

**Now my last question:**
Why were the libraries for the Tasmota versions <= 8.x? removed from Tasmota. So far there have been no problems with the parallel libraries for the core. This can hardly be a storage space problem, because GitHub is quite large.

**Reply by Jason2866** [reply]
This was at the time where we uploaded the platforms/frameworks to Platformios Registry.
This is NOT github. Platformio removed old stuff. Not anything we can influence.
Since a few years we do not use Platformio Registry. We store everything in Github.

**Reply by wchulze** [reply]
Mhm, strange. I have run all versions in parallel on another Windows machine with the current platformIO (3.3.3). However, this one did not see any Tasmota version greater than 11.x.
I think I noticed on the current machine that after I looked at version 14.2.0 an update took place without me doing anything. Since I back up my machine, I was able to roll back. Now all versions are there again. I will repeat the process in the next few days.

**Reply by Jason2866** [reply]
If the have been local installed at some day they are there and can be used. The problem arrives when a fresh Platformio setup is done.

**Reply by wchulze** [reply]
Since I can restore my system quite easily, I have, as promised, carried out some investigations.

You wrote in the description of version  14.1.0

**Support of ESP8266 Core versions before 2.7.6 and ESP32 Core versions before 3.0.0 have been removed.**

This seems to be a very serious problem. After first viewing the version 14.1.0 in in the PlatformIO without compiling, I was no longer able to compile my older Tasmota versions (e.g. 8.5.1). As a result, it crashed my compilation environment. This is very curious.

The older versions of Tasmota do not behave like this. Nothing is deleted, it runs stable
 
In all the cases described, the version of the PlatformIO was not changed. It seems to me that Tasmota is the trigger that may have instructed the PlatformIO to carry out the deletion. Please change this behavior.

**Comment by Jason2866** [comment]
> In all the cases described, the version of the PlatformIO was not changed. It seems to me that Tasmota is the trigger that may have instructed the PlatformIO to carry out the deletion. Please change this behavior.

We are using Platformio as build system. We are not the maintainers. It is like it is. Platformio **has** changed since that time. Old versions of Platformio do not support actual espressif MCUs like the esp32-s3

**Reply by wchulze** [reply]
Of course you can't change the behavior of PlatformIO. It's clear that PlatformIO changes. Everything was determined with the current version 3.3.3.

Since I also use the system for other projects, I'm sure that PlatformIO doesn't make any independent changes to the settings!

I hope there are no side effects of Tasmota on my projects. From now on I will only use the Tasmota development environment in a special virtual machine.
